src/lib/Cropper.svelte

```diff
@@ -229,34 +230,55 @@
       restrictPosition
     )
 
-    dispatch('cropcomplete', {
+    oncropcomplete?.({
       percent: croppedAreaPercentages,
       pixels: croppedAreaPixels,
     })
   }
 
   // ------ Reactive statement ------
   //when aspect changes, we reset the cropperSize
-  $: if (imgEl) {
-    cropperSize = cropSize ? cropSize : helpers.getCropSize(imgEl.width, imgEl.height, aspect)
-  }
+  $effect(() => {
+    if (imgEl) {
+      cropperSize = cropSize ? cropSize : helpers.getCropSize(imgEl.width, imgEl.height, aspect)
+    }
+  })
 
   // when zoom changes, we recompute the cropped area
-  $: zoom && emitCropData()
+  let lastZoom = $state(zoom);
```

Comment:
We should not need to track the last zoom value ourselves, Svelte should do it automatically as long as we access `zoom` in the effect.

Reply:
This one can be removed safely without causing loops. Just added it for testing and forgot to remove